llvm/test/CodeGen/DirectX/scalar-data.ll

```diff
@@ -1,4 +1,5 @@
-; RUN: opt -S -passes='dxil-data-scalarization,function(scalarizer<load-store>),dxil-op-lower' -mtriple=dxil-pc-shadermodel6.3-library %s | FileCheck %s
+; RUN: opt -S -passes='dxil-data-scalarization,function(scalarizer<load-store>),dxil-op-lower' -mtriple=dxil-pc-shadermodel6.3-library %s | FileCheck %s -check-prefix DATACHECK
+; RUN: opt -S -passes='dxil-data-scalarization,dxil-flatten-arrays,function(scalarizer<load-store>),dxil-op-lower' -mtriple=dxil-pc-shadermodel6.3-library %s | FileCheck %s
```

Comment:
The problem is two fold I want this to be a scalarizer test case, but  we are using the `llc`  run line. `llc` was important to find bugs were my passes  could have created invalid llvmir for later passes.  Dropping a flatten-arrays pass also means dropping the llc run line. So I saw three paths forward. 

1.  Drop llc and make this a scalarizer only test. Downside if later passes break things the CI won't catch it.
2. Drop the scalarizer specifics from the test. Downside is we don't have scalarizer only tests
3. Do what I'm currently doing to maintain both llc and data scalarization specifics in one test.

Reply:
So I think what we should do is add a test file for scalarization that checks that it does the right thing for non-flattened arrays. This will only have an `opt` run line.

Then we have two options for the existing scalarization tests that have `llc` run lines:
1. Update them to just have flattened arrays, since that's what we'll actually see in practice
2. Update the `opt` lines to include dxil-flatten-arrays

I think I have a preference for (1) since that makes it much easier to debug if/when the tests fail.